src/sdk/sdk.interface.ts

```diff
@@ -22,6 +23,7 @@ export interface DialectSdk {
   readonly threads: Messaging;
   readonly dapps: Dapps;
   readonly wallet: Wallets;
+  readonly nameService: NameService;
```

Comment:
Hm, in my mind it should be part of the FindThreadQuery

Reply:
But it can be used not only for thread in theory right? I thought about it like this - whenever you need replace you PK you can call function and get result.

Reply:
hm, good point. But anyway I think it'll be cool to have as a high level api too and add it to find thread query. @tsmbl wdyt?

Reply:
So, my opinion it that I think it should be configurable in sdk in moreless same way as backends and should affect both thread lookup and message info

Also I'm a bit confused that name service is exposed directly from dialect sdk, since name services are external systems, why they should be a part of dialect sdk?

Basically there're 2 use-cases for name services, that I see
1. W/o sdk itself just a separate libraries for name lookups:  solved by creating libs e.g.  @dialectlabs/name-service (for interface only) @dialectlabs/name-service-cardinal, @dialectlabs/name-service-sns, etc
2. With sdk i.e. in messaging: this requires more tight and fluent integration with existing apis and based on (1)

I would propose to start with (1): just providing pluggable libs for name services event w/o sdk integration -- this will allow us to use it e.g. to power slope push notifications and proceed with (2) after (1) is done

Wdyt?

Reply:
@AlekseiPetrovDev @zhelezkov we can also make a short call today to discuss it